src/openqaoa-braket/backends/qaoa_braket_qpu.py

```diff
@@ -46,32 +48,33 @@ class QAOAAWSQPUBackend(
         used. This is False by default. Not all providers provide this feature.
     """
 
-    def __init__(
-        self,
-        qaoa_descriptor: QAOADescriptor,
-        device: DeviceAWS,
-        n_shots: int,
-        prepend_state: Optional[Circuit],
-        append_state: Optional[Circuit],
-        init_hadamard: bool,
-        cvar_alpha: float,
-        qubit_layout: List[int] = [],
-        disable_qubit_rewiring: bool = False,
-    ):
-
-        QAOABaseBackendShotBased.__init__(
-            self,
-            qaoa_descriptor,
-            n_shots,
-            prepend_state,
-            append_state,
-            init_hadamard,
-            cvar_alpha,
-        )
+    def __init__(self,
+                 qaoa_descriptor: QAOADescriptor,
+                 device: DeviceAWS,
+                 n_shots: int,
+                 prepend_state: Optional[Circuit],
+                 append_state: Optional[Circuit],
+                 init_hadamard: bool,
+                 cvar_alpha: float,
+                 disable_qubit_rewiring: bool = False,
+                 initial_qubit_mapping: Optional[List[int]] = None):
+
+        QAOABaseBackendShotBased.__init__(self,
+                                          qaoa_descriptor,
+                                          n_shots,
+                                          prepend_state,
+                                          append_state,
+                                          init_hadamard,
+                                          cvar_alpha,
+                                        )
         QAOABaseBackendCloud.__init__(self, device)
 
-        self.qureg = self.qaoa_descriptor.qureg
-        self.qubit_layout = self.qureg if qubit_layout == [] else qubit_layout
+        self.qureg = list(range(self.n_qubits))
+        self.problem_reg = self.qureg[0:self.problem_qubits]
```

Comment:
Incidentally, is `n_qubits` inferred or specified by the user?

And if inferred, is it the size of the set of indexes in the terms or is it the largest index?

Reply:
`n_qubits` is, normally, obtained through the Cost Hamiltonian which is a `Hamiltonian` Object. The number of qubits the `Hamiltonian` object returns is inferred from the hamiltonian of the problem based on the 2-qubit pauli terms.

Reply:
`n_qubits` can not be obtained through the cost hamiltonian anymore, because the qubit routing could use more qubits than `n_qubits_problem`. So `n_qubits` has to be "all the qubits used by the routing".

Reply:
Its an either or for now I believe. If there is no routing algorithm. `n_qubits` is still obtained from the cost hamiltonian.

Reply:
@shahidee44 is right.